keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -198,16 +283,48 @@ Feature is already in Beta.
 
 #### Beta -> GA Graduation
 
-- 3 examples of real world usage
-- support for remote TLS handshakes (e.g. TPM/KMS-hosted keys)
+- Three examples of real world usage
+  + Confirm interactive and non-interactive UX is acceptable
+  + Confirm no hacks are being performed to workaround limitations
+  + Confirm that configuration of plugin
+    * Is correctly handled
+    * Is well-supported by the `kubeconfig` file format
+- Create the `client.authentication.k8s.io/v1` `ExecCredential` struct
+- Address known bugs and add tests to prevent regressions
+- Docs are up-to-date with latest version of APIs
+- Docs describe set of best practices (i.e. do not mutate `kubeconfig`)
+
+Question: does this need conformance tests?  What would such a test look like?
+
+### Upgrade / Downgrade Strategy
+
+The distribution of executables to end users for use with clients is out of the
+scope of this KEP.  Thus end users are responsible for confirming that the
+executable they are attempting to use is compatible with `exec.apiVersion`.
+
+### Version Skew Strategy
+
+The client is aware of its configured `exec.apiVersion`.  It must validate that
+the status response from the executable has the matching API version to prevent
+it from misinterpreting the response.
+
+## Drawbacks
+
+End users must take care to only use trusted executables as they could easily
+read sensitive data from the user's file system.
+
+A maliciously crafted `kubeconfig` file could be used to execute arbitrary
```

Comment:
We could do more to make this harder.

Reply:
I have not added any changes to the KEP at this point to address this (per my understanding from yesterday's sig-auth meeting).

Reply:
Quick and crazy brainstorm:
* Verify that the execution options are not controlled by untrusted users. This validation would be similar to what https://github.com/google/path-auditor preforms.
  * Verify that the kubeconfig is only under the control of the exec'ing user and/or root.
  * Verify that the exec'd binary is only under the control of the exec'ing user and/or root.
* Add option to verify that the exec'd binary matches some hash. sudoers configuration allows this. A more complex alternative is to add support for code signing.
* Explore cheap sandboxing options of the exec'd process.